### PR TITLE
Add tests for signals

### PR DESCRIPTION
**Pull request by Maran23**
Add tests for signals in `Object`.
The following test cases are added:
- Adding signals
- Emitting signals

### CODE REVIEW

**Review by Geometror — APPROVED** [review]
Looks fine to me!

**Review by deralmas — COMMENTED** [review]
I pointed some english-related nitpicks. As usual I'm not a native speaker so I might be wrong.

Other than those, this looks fine!

File: tests/core/object/test_object.h [hunk 1]
```diff
@@ -281,6 +281,84 @@ TEST_CASE("[Object] Absent name getter") {
...
+		ERR_PRINT_ON;
+		CHECK(object.has_signal("my_custom_signal"));
+
+		List<MethodInfo> signals_after_existing_added;
+		object.get_signal_list(&signals_after_existing_added);
+
+		CHECK_EQ(signals_after.size(), signals_after_existing_added.size());
+	}
+
+	SUBCASE("Try adding a preexisting signal should not have any effect") {
```
[the remark this answers is not in the document]
Comment (Maran23) [reply tests/core/object/test_object.h]: done!

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**